src/main/scala/guessbook/GuessBookUserInput.scala

```diff
@@ -0,0 +1,225 @@
+          game.printRemainingBooks(booksOnBoard(guesserIndex))
+          // end of turn
+          playerTurns(selectedBooks, booksOnBoard, guesser = guesser % 2 + 1)
+        }
+        case 2 => { // guessing book title
+          val guessedTitle: String = specifyBookTitle()
+          val guessResult: (Seq[Book], Boolean) = game.guessBookTitle(selectedBooks(guesserIndex), guessedTitle, guesser, booksOnBoard(guesserIndex))
+          booksOnBoard(guesserIndex) = guessResult._1
+          if (guessResult._2)  // player guessed correctly
+            println("END OF GAME")
+          else {
+            println(s"Books remaining on P$guesser's board:")
+            game.printRemainingBooks(booksOnBoard(guesserIndex))
+            playerTurns(selectedBooks, booksOnBoard, guesser = guesser % 2 + 1)
+          }
+        }
+      }
+    }
+  }
+
+  def selectBookForOpponent(guesser: Int, numBlankLines: Int = 25): Book = {
+    val nonGuesser = guesser % 2 + 1
+    println(s"Player $guesser, select a book for Player $nonGuesser to guess.")
+    println(s"Player $nonGuesser, please look away.\n")
+    println("The available books are:")
+    game.printRemainingBooks(game.allBooks)
+    println()
+
+    @tailrec
+    def getBookInGame(nonGuesser: Int): Book = {
+      println(s"Enter the title of the book for Player $nonGuesser to guess:")
+      val titleInput = StdIn.readLine()
+      val matchingBook = game.allBooks.find(book => book.title.toLowerCase == titleInput.toLowerCase)
+      matchingBook match {
+        case Some(book) => {
+          println("Book selected." + "\n" * numBlankLines)
+          book
+        }
+        case None => {
+          println(s"$titleInput is not the title of a book in the game. Please try again.\n")
+          getBookInGame(nonGuesser)
+        }
+      }
+    }
+
+    getBookInGame(nonGuesser)
+  }
+
+  @tailrec
+  def chooseGameMode(): Int = {
+    println("Enter [1] for each player's selected book to be drawn randomly\n" +
+      "Enter [2] for each player to choose the book for their opponent to guess")
+
+    val input = StdIn.readLine()
+    Try(input.toInt) match {
+      case Success(x) if (x == 1 || x == 2) => x
+      case _ => {
+        println("Invalid input. Please try again.\n")
+        chooseGameMode()
+      }
+    }
+  }
+
+  @tailrec
+  def chooseAttributeOrTitle(): Int = {
+    println("Enter [1] to guess an attribute value\n" +
+      "Enter [2] to guess the title")
+
+    val input = StdIn.readLine()
+    Try(input.toInt) match {
+      case Success(x) if (x == 1 || x == 2) => x
+      case _ => {
+        println("Invalid input. Please try again.\n")
+        chooseAttributeOrTitle()
+      }
+    }
+  }
+
+  def specifyBookTitle(): String = {
+    println("Which do you think is the selected book?")
+    val titleInput = StdIn.readLine()
+    val matchingTitle: Option[String] = game.allTitles.find(title => title.toLowerCase == titleInput.toLowerCase)
+    matchingTitle match {
```

Comment:
if you inline matchingTitle here then this could be tail recursive. I think it's ok to inline this since matchingTitle as a val isn't tooooooo complex, but ofc it's a trade off, so don't do this if you think it's better to have matchingTitle separate